src/typings/assets.d.ts

```diff
@@ -0,0 +1,22 @@
+type DotExtendedString<T extends string> = `${T}.`;
```

Comment:
Ok, I reviewed this again, this time more thoroughly.. 
Since the TS utils are pretty complicated to understand at first glance I'd suggest to make them more readable. 

1. I'd change params names like `T` and `K` to something with more meaning.. 
for example, maybe change this `AssetRecord<T extends string, K extends object = {}, This extends object>` to `AssetRecord<PATH extends string, ASSETS extends object = {}, ASSETS_CLASS extends object>`
2. Let's add a short description above the complicated utils of what exactly they do. If we'll have to investigate an issue in the future it can be very useful 
3. RecRecord util's name is not so clear, can you think of a better name?